status/status_ext_test.go

```diff
@@ -203,3 +205,32 @@ func (s) TestStatusDetails(t *testing.T) {
 		})
 	}
 }
+
+// TestStatus_ErrorDetailsMessageV1 verifies backward compatibility of the
+// status.Details() method when using protobuf code generated with only the
+// MessageV1 API implementation.
+func (s) TestStatus_ErrorDetailsMessageV1(t *testing.T) {
+	details := []protoadapt.MessageV1{
+		&tpb.SimpleMessage{
+			Data: "abc",
+		},
+		&testpb.Empty{},
+	}
+	s, err := status.New(codes.Aborted, "").WithDetails(details...)
+	if err != nil {
+		t.Fatalf("(%v).WithDetails(%+v) failed: %v", s, details, err)
+	}
+	got := s.Details()
+	gotDetails := []protoadapt.MessageV1{}
+	for i := range got {
+		msg, ok := got[i].(protoadapt.MessageV1)
+		if !ok {
+			t.Fatalf("(%v).Details() returned message that doesn't implement protoadapt.MessageV1: %v", s, got[i])
+		}
+		gotDetails = append(gotDetails, msg)
+	}
+
+	if diff := cmp.Diff(details, gotDetails, protocmp.Transform()); diff != "" {
```

Comment:
I just have glance on `cmp.Diff`, look like it is used to convert `proto.Message` into map-like structure, then perform a recursive "value-equal" check.
I think such examination does not satisfy "the proto message returned by Details() is of the same type that was given to WithDetails()". A `reflect.Type` check is preferred in such situation.